### PR TITLE
feat: collapsible categories

### PR DESCRIPTION
**Pull request by tsmrachel**
## Motivation

#1084 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Tested against desktop browsers 

- Microsoft Edge 42.17134.1.0
- Chrome Version 70.0.3538.110 (Official Build) (64-bit)

Test Plan 

1. Tested that it's working without collapsible menu enabled
2. Added in subcategories and tested that both subcategories and links (under the category) is able to be collapsed/expanded
3. Tested against full width and minimal width on desktop to ensure that UI looks fine
4. Tested that the category remains expanded after navigating to a new page (link under subcategory) within the sidebar

image

image

## Related PRs

Null

### CODE REVIEW

**Comment by tsmrachel** [comment]
@endiliey done! will need to remove/revert commit: ebf7ad8 if everything is fine

File: v1/lib/core/nav/SideNav.js [hunk 1]
```diff
@@ -133,6 +146,29 @@ class SideNav extends React.Component {
         <script
           dangerouslySetInnerHTML={{
             __html: `
+            var coll = document.getElementsByClassName('collapsible');
```
Comment (mokimo) [inline v1/lib/core/nav/SideNav.js]: Heyhey, is there any particular reason you're using var in favor of const/let?

Comment (yangshun) [reply v1/lib/core/nav/SideNav.js]: The code within `<script>` is shipped as-is to browsers, and not all browsers understand `let`. We stick to ES5 in such situations.

Comment (mokimo) [reply v1/lib/core/nav/SideNav.js]: Thx for the clarification!

**Pull request merged**